server/src/util/sourcing.ts

```diff
@@ -0,0 +1,92 @@
+import * as path from 'path'
+import * as LSP from 'vscode-languageserver'
+import * as Parser from 'web-tree-sitter'
+
+import { untildify } from './fs'
+
+// Until the grammar supports sourcing, we use this little regular expression
+const SOURCED_FILES_REG_EXP = /^(?:\t|[ ])*(?:source|[.])\s*(\S*)/gm
+
+export function getSourcedUris({
+  fileContent,
+  fileUri,
+}: {
+  fileContent: string
+  fileUri: string
+}): Set<string> {
+  const uris: Set<string> = new Set([])
+  let match: RegExpExecArray | null
+
+  while ((match = SOURCED_FILES_REG_EXP.exec(fileContent)) !== null) {
+    const relativePath = match[1]
```

Comment:
if user has:

```sh
source "./my_file.sh"
```

Then `relativePath` will contain quotes around it, and the sourcedUri that gets added has quotes at the end of the absolute path, and thus will fail to be read later